site/content/entry/yuzu-progress-report-aug-2020/index.md

```diff
@@ -0,0 +1,209 @@
+
+<!--more-->
+
+## Project Mjölnir
+
+### Part 1 - Input Rewrite
+
+Let's start with the biggest change. One of yuzu's weakest aspects was its input support. It was riddled with bugs, not very intuitive, and lacked several critical features. Earlier this year, [jroweboy](https://github.com/jroweboy) and [Rei](https://github.com/Its-Rei) decided to start working on what ended up being a rewrite of the whole input UI and its internal working dubbed Project Aphrodite.
+
+{{< imgs
+    "./oldinput.png| Never gonna give you up"
+  >}}
+  
+As you can see in the following images, the original draft lacked some of the features we have working right now and some that we plan to add in the future. No code is ever completed, better ideas and new implementations can always be achieved. So that's exactly what happened, Rei continued to iterate on the concept design that is inspired by [RPCS3's](https://github.com/RPCS3/rpcs3) controller UI.
+
+{{< imgs
+    "./concept.png| The first draft, in all its glory"
+  >}}
+
+With this concept in mind, jroweboy implemented an automapping feature which correctly maps all the inputs upon selecting an input device.
+
+{{< imgs
+    "./automap.mp4| Finally, no more spending hours mapping controls!"
+  >}}
+
+Since jroweboy's departure from emulator development, [Morph](https://github.com/Morph1984) continued his legacy and picked up where he left off not wanting to have all this work go to waste. With renewed hope, both he and [Rei](https://github.com/Its-Rei) are continuously finding and fixing bugs and refining the UI as much as possible to achieve the best user experience across all three UI themes present in yuzu, which culminated into the UI we have today.
+
+{{< single-title-imgs
+    "The fruits of our labor, the triforce of themes is complete!"
+    "./light.png"
+    "./dark.png"
+    "./midnightblue.png"
+  >}}
+
+The user experience would not be complete unless a [long standing issue where updating controllers in the UI will not update them in game](https://github.com/yuzu-emu/yuzu/issues/2906) is fixed. Knowing this, Morph decided to find out what is causing this issue and fix it along with the UI changes. With some guidance from the shark [ogniK](https://github.com/ogniK5377), he discovered that 1. The frontend was not signalling to the backend when a controller changes configuration and 2. `StyleSetChangedEvents` were not being signalled on a controller change. By fixing these 2 issues, he managed to fix input issues in a wide array of games and allowing users to change their controller and input devices while in game without needing to restart yuzu.
+
+{{< imgs
+    "./controllerselect.mp4| Smash players rejoice!"
+  >}}
+
+The games that demonstrated the most benefit from these fixes are Pokemon Let's Go: Pikachu and Pokemon Let's Go: Eevee as single joycons can now be properly detected in game, bringing us one step closer to emulation accuracy.
```

Comment:
In previous articles, we established that game titles should be emphasized by being put in code tags: `